lib/active_job/uniqueness/strategies/base.rb

```diff
@@ -86,6 +96,12 @@ def handle_conflict(on_conflict:, resource:, event: :conflict)
           end
         end
 
+        def handle_error(resource:, on_error:, error:)
+          raise error unless on_error
+
+          on_error.call(job, resource:, error:)
```

Comment:
This syntax is not available in Rubies pre-3.1

Reply:
thanks for the fast feedback :) will change.

You've been fast, in the meantime i've changed the naming a bit and added support for a global config